common/src/main/kotlin/io/homeassistant/companion/android/common/data/HttpExceptionCallAdapterFactory.kt

```diff
@@ -0,0 +1,53 @@
+package io.homeassistant.companion.android.common.data
+
+import io.homeassistant.companion.android.common.exception.HttpException
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+
+class HttpExceptionCallAdapterFactory : CallAdapter.Factory() {
```

Comment:
From https://developers.home-assistant.io/docs/android/best_practices#documentation `Future-proof: Ask yourself, "Will I understand what I did in 6 months?"`

I think we would benefits from having a documentation on the class definition explaining what is the goal of this factory.